@kfisler's dcic2024.arr context file added

ds26gte opened the pull request:
Kathi supplied this context file for use with DCIC.

ds26gte committed:
@kfisler's dcic2024.arr context file added

blerner commented:
The build failed due to unbound identifiers...

ds26gte committed:
dcic2024.arr: use context starter2024

ds26gte commented:
> The build failed due to unbound identifiers...

FIxed by explicitly using context starter2024. OK to merge?

blerner commented:
This is likely not quite what you mean to do -- right now this context _only_ provides 26 table functions, and nothing else.  Pyret doesn't implicitly re-provide imported things (this is analogous to `require` and `provide` in Racket). If you're trying to build atop `starter2024`, then you probably need to

```
use context empty
import starter2024 as S
provide from S: *, type *, data * end
include from S: *, type *, data * end

...rest as written...
```
You also have no contracts on any of the functions (they're in comments only), so you'll get no annotation checking on them at all.

I'd recommend trying to use this with some DCIC programs to see if it actually works for them as intended, and you'll pretty quickly find the missing exports.

ds26gte committed:
Merge branch 'horizon' into dcic2024

ds26gte committed:
- dcic2024: use essentials2024 for base context
- (starter2024 has problems w various aliased functions)
- don't import lists as L as default context already provides it
